indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -84,6 +84,19 @@ public WorkerTaskMonitor(
   public void start()
   {
     try {
+      // cleanup any old running task announcements which are invalid after restart
+      for (TaskAnnouncement announcement : workerCuratorCoordinator.getAnnouncements()){
```

Comment:
the getAnnouncements() doesn't reads from the cache, so its fine to not wait for the cache to be initialized here.